db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerImpl.java

```diff
@@ -894,6 +894,24 @@ public boolean nextJoinOrder() throws StandardException{
                 }
             }
 
+            // If we want to re-use this Optimizable without re-planning it, return now.
+            ResultSetNode optimizableResultSet = (ResultSetNode) optimizableList.getOptimizable(nextOptimizable);
+            if (optimizableResultSet.skipBindAndOptimize) {
+                if (optimizableResultSet instanceof FromTable) {
+                    FromTable fromTable = (FromTable)optimizableResultSet;
+                    AccessPath bestAP = fromTable.getBestAccessPath();
+                    fromTable.startOptimizing(this,currentRowOrdering);
+                    if (nextOptimizable == proposedJoinOrder[0] )
+                        bestAP.setCostEstimate(fromTable.getTrulyTheBestAccessPath().getCostEstimate());
+                    AccessPath currentAP = fromTable.getCurrentAccessPath();
+                    CostEstimate costEstimate=fromTable.getCostEstimate(this);
+                    costEstimate.setCost(Double.MAX_VALUE,Double.MAX_VALUE,Double.MAX_VALUE);
+                    currentAP.setCostEstimate(costEstimate);
+                }
```

Comment:
Can you please add more comment about under which condition we enter this code path?

Reply:
OK, added the comment:

>             // If we want to re-use this Optimizable without re-planning it, return now.
>             // This is the case where nextOptimizable uses a Unioned Index Scans access path,
>             // and it is being joined with another table which itself might use a UIS access path.
>             // We want to set the join cost estimate to an uninitialized state and set assignedTableMap
>             // to include the proper tables, but we don't want to push any new predicates to this
>             // table because it has already been materialized into a sequence of statements that
>             // can't be modified.